geonode/api/resourcebase_api.py

```diff
@@ -525,6 +526,25 @@ class Meta(CommonMetaApi):
         resource_name = 'maps'
 
 
+class MapLayersResource(CommonModelApi):
+
+    """Includes everything from Maps API plus details for each layer in map"""
+
+    class Meta(CommonMetaApi):
+        queryset = Map.objects.distinct().order_by('-date')
+        if settings.RESOURCE_PUBLISHING:
+            queryset = queryset.filter(is_published=True)
+        resource_name = 'maplayers'
+
+    def dehydrate(self, bundle):
```

Comment:
Good thinking, @simod.

Here are some performance test results for this new API endpoint. These tests were run against a virtual machine with 4 CPU cores and 8gb of RAM. I can run the same tests on real hardware if needed but it will take more time.

I created some example maps with 10, 20, 50, and 100 layers. For each of these, I accessed the new endpoint at `/api/maplayers/<map-id>` and:

- Took the average of ten timed `wget` requests while GeoNode was sitting idle.
- Sent requests to the endpoint in rapid succession with jMeter and took the average after 500 requests (plenty of time for the average to stabilize).

Here are the results:

*10 layers*
Idle: 0.10 seconds
jMeter: 0.51 seconds

*20 layers*
Idle: 0.18 seconds
jMeter: 0.69 seconds

*50 layers*
Idle: 0.26 seconds
jMeter: 1.29 seconds

*100 layers*
Idle: 0.46 seconds
jMeter: 2.29 seconds

If you access the new endpoint without specifying a map ID (just `/api/maplayers/`) it loads the same thing that `/api/maps/` does, so there is no performance impact there.